nav2_util/include/nav2_util/string_utils.hpp

```diff
@@ -23,22 +25,35 @@ namespace nav2_util
 
 typedef std::vector<std::string> Tokens;
 
-/*
- * @brief Remove leading slash from a topic name
+/** @brief Remove leading slash from a topic name
  * @param in String of topic in
  * @return String out without slash
 */
 std::string strip_leading_slash(const std::string & in);
 
-///
-/*
- * @brief Split a string at the delimiters
+/** @brief Split a string at the delimiters
  * @param in String to split
  * @param Delimiter criteria
  * @return Tokens
 */
 Tokens split(const std::string & tokenstring, char delimiter);
 
+/** @brief Parse a vector of vector of floats from a string.
+ * @param input
+ * @param error_return
+ * Syntax is [[1.0, 2.0], [3.3, 4.4, 5.5], ...] */
+std::vector<std::vector<float>> parseVVF(const std::string & input, std::string & error_return);
+
+/**
+ * @brief Make the vector from the given string.
+ * @param str_pts
+ * @param vector_pts
+ * Format should be bracketed array of arrays of floats, like so: [[1.0, 2.2], [3.3, 4.2], ...]
+ *
+ */
+bool makeVectorPointsFromString(const std::string & safety_zone_str,
```

Comment:
These variable names should be more general for the general approach that this function is. Its just a param string in and a set of points out